### PR TITLE
feat: migrate to mysql + setup https configs

### PR DESCRIPTION
**Pull request by aidanaden**
renamed version of #28

### CODE REVIEW

File: backend/src/main.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
 import { ConfigService } from "@nestjs/config";
 import { NestFactory } from "@nestjs/core";
 import helmet from "helmet";
-import * as csurf from "csurf";
+import * as path from "path";
 import * as cookieParser from "cookie-parser";
+import * as fs from "fs";
+import * as csurf from "csurf";
```
Comment (ngjunkang) [inline backend/src/main.ts:7]: can remove line 4, 6 and 7 (unused)

File: frontend/src/components/auth.tsx [hunk 2]
```diff
@@ -2,6 +2,9 @@ import { useEffect, useState } from "react";
 import io from "socket.io-client";
 import type { Socket } from "socket.io-client";
 
+const serverUrl = "http://localhost:5000";
+const websocketUrl = "ws://localhost:5000";
```
Comment (ngjunkang) [inline frontend/src/components/auth.tsx:6]: hardcoded? just a reminder to put env var

**Review by ngjunkang — APPROVED** [review]
LGTM, just some nits

File: frontend/src/components/auth.tsx (hunk 2 above)
Comment (aidanaden) [reply frontend/src/components/auth.tsx:6]: j for testing purposes - once start proper will create the relevant env variables!

**Review by yusufaine — APPROVED** [review]
LGTM, backend still works after the db change!

**Pull request merged**